atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -218,11 +218,11 @@ public List<OpenTransaction> startTransactions(List<? extends PreCommitCondition
             openTransactionCounter.inc(transactions.size());
             return transactions;
         } catch (Throwable t) {
+            responses.forEach(response -> lockWatchManager.removeTransactionStateFromCache(
+                    response.startTimestampAndPartition().timestamp()));
```

Comment:
Moved upwards so that, if we throw during `tryUnlock`, we still remove from the cache

Reply:
Not overall actionable: If you're worried about `tryUnlock` throwing, I think we want more than this (we'd lose `t`). 

Though seriously absent bug in our product code `tryUnlock` shouldn't throw